src/sentry/users/api/endpoints/user_emails.py

```diff
@@ -127,27 +152,35 @@ def post(self, request: Request, user: User) -> Response:
         result = validator.validated_data
         email = result["email"].lower().strip()
 
-        try:
-            new_useremail = add_email(email, user)
-        except DuplicateEmailError:
-            new_useremail = user.emails.get(email__iexact=email)
+        use_signed_urls = options.get("user-settings.signed-url-confirmation-emails")
+        if use_signed_urls:
+            add_email_signed(email, user)
             return self.respond(
-                serialize(new_useremail, user=request.user, serializer=UserEmailSerializer()),
-                status=200,
-            )
-        else:
-            logger.info(
-                "user.email.add",
-                extra={
-                    "user_id": user.id,
-                    "ip_address": request.META["REMOTE_ADDR"],
-                    "email": new_useremail.email,
-                },
-            )
-            return self.respond(
-                serialize(new_useremail, user=request.user, serializer=UserEmailSerializer()),
+                {"email": _("A verification email has been sent. Please check your inbox.")},
```

Comment:
what does this look like on the frontend?

Reply:
This pops up as a toast in the bottom right corner of the frontend UI. 
I'll get the frontend PR created. :)